core/commands/blame.py

```diff
@@ -238,9 +274,132 @@ def run(self, cursor_pt, edit):
+            except Exception:
+                return 1
+
+    def open_commit(self):
         # Uncommitted blocks.
-        if not short_hash.strip():
+        commit_hash = self.selected_commit_hash()
+        if not commit_hash:
             return
 
-        self.view.window().run_command("gs_show_commit", {"commit_hash": short_hash})
+        self.view.window().run_command("gs_show_commit", {"commit_hash": commit_hash})
+
+    def commit_before(self, position, commit_hash):
+        # I would like it to be something like this, but I could make it work when
+        # i reached the end of the end second last commit
+        # previous_commit_hash =  self.git("show", "--format=%H", "--no-patch", "{}^-1".format(commit_hash)).strip()
+
+        log_commits = self.git("log", "--format=%H", "--follow", "--", self.file_path).strip()
+        log_commits = log_commits.split("\n")
+
+        commit_hash_len = len(commit_hash)
+
+        for idx, commit in enumerate(log_commits):
+            if commit.startswith(commit_hash) :
+                if position == "older":
+                    if idx < len(log_commits)-1:
+                        return log_commits[idx+1]
+                    else:
+                        # if we are at the end display this the oldest commit
+                        return commit
+                elif position == "newer":
+                    return log_commits[idx-1]
+
+    def newst_commit_for_file(self):
+        return self.git("log", "--format=%H", "--follow", "-n", "1", self.file_path).strip()
+
+    def find_line_and_open(self):
+        commit_hash = self.selected_commit_hash().strip()
+        if not commit_hash:
+            self.view.settings().set("git_savvy.commit_hash", self.newst_commit_for_file())
+        else:
+            self.view.settings().set("git_savvy.commit_hash", self.commit_before("older", commit_hash))
+
+        self.view.run_command("gs_blame_initialize_view")
+
+    def open(self, position):
+        settings = self.view.settings()
+        commit_hash = settings.get("git_savvy.commit_hash")
+        if not commit_hash:
+            settings.set("git_savvy.commit_hash", self.newst_commit_for_file())
+        else:
+            previous_commit_hash = self.commit_before(position, settings.get("git_savvy.commit_hash"))
+            settings.set("git_savvy.commit_hash", previous_commit_hash)
+        self.view.run_command("gs_blame_initialize_view")
+
+    def show_file_at_commit(self, from_line=False):
+        if from_line:
+            commit_hash = self.selected_commit_hash() or 'HEAD'
+        else:
+            commit_hash = self.view.settings().get("git_savvy.commit_hash", "HEAD")
+
+        self.view.window().run_command("gs_show_file_at_commit", {
+            "commit_hash": commit_hash,
+            "filepath": self.file_path,
+            "lineno": self.find_lineno(),
+            "lang" : self.view.settings().get('git_savvy.original_syntax', None)
```

Comment:
We could also capture the scopes of the file in the output panel to highlight the blame view.